### PR TITLE
Remove the remainder of the appdomain unload code.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: src/mono/mono/metadata/appdomain.h [hunk 1]
```diff
@@ -94,9 +94,6 @@ mono_domain_unload (MonoDomain *domain);
 MONO_API void
 mono_domain_try_unload (MonoDomain *domain, MonoObject **exc);
 
-MONO_API mono_bool
-mono_domain_is_unloading   (MonoDomain *domain);
```
Comment (lambdageek) [inline src/mono/mono/metadata/appdomain.h:98]: This is a public header.  Need to keep a `MONO_RT_EXTERNAL_ONLY` symbol that always returns false

File: src/mono/mono/metadata/exception.h [hunk 2]
```diff
@@ -129,12 +129,6 @@ MONO_API MONO_RT_EXTERNAL_ONLY
 MonoException *
 mono_get_exception_synchronization_lock (const char *msg);
 
-MONO_API MonoException *
-mono_get_exception_cannot_unload_appdomain (const char *msg);
-
-MONO_API MonoException *
-mono_get_exception_appdomain_unloaded (void);
```
Comment (lambdageek) [inline src/mono/mono/metadata/exception.h:136]: These are public too.

**Review by lambdageek — APPROVED** [review]
Keep the public API methods, mark them with `MONO_RT_EXTERNAL_ONLY` and make them return NULL, FALSE, etc

**Review by CoffeeFlux — APPROVED** [review]
This will conflict with some of my local stuff where I removed this too, but I'll rebase. LGTM.

**Review by lambdageek — CHANGES_REQUESTED** [review]
API funcs

**Review by lambdageek — APPROVED** [review]

**Pull request merged**